vectorstores/options.go

```diff
@@ -44,3 +48,17 @@ func WithEmbedder(embedder embeddings.Embedder) Option {
 		o.Embedder = embedder
 	}
 }
+
+// WithID returns an Option for setting vectorId for document.
+// Inserting documents using a custom vector identifier or auto generated UUID.
+// This approach is beneficial when we need to perform deletions or updates on the vector within the Pinecone index.
+func WithID(id string) Option {
+	if id != "" {
+		return func(o *Options) {
+			o.ID = id
+		}
+	}
+	return func(o *Options) {
+		o.ID = uuid.NewString()
+	}
```

Comment:
This logic should not be in the option, but in the AddDocuments method, because we need to create and UUID for each of the documents upserted if no IDs are specified and we dont know how many documents will be upserted in this function.